artifact_package resource

ivey opened the pull request.

ivey committed:
first pass artifact_package

KAllan357 commented on providers/package.rb:
```diff
@@ -0,0 +1,47 @@
+  end
+
+  file tempfile do
+    action :delete
```
Will this mean that every time this resource runs (if the resource itself is not protected by some logic) that wherever we use this resource, we'll be downloading the artifact and deleting it, even if we don't install it again?

ivey replied:
Addressed.

ivey committed:
Make caching work

Don't delete the downloaded package, and include a SHA of the location
string in the downloaded file name.

ivey committed:
Make it so the cache file can be used by yum

ivey committed:
readme updated

KAllan357 commented:
:+1:

ivey committed:
s/create/install/ because Kyle is super picky

ivey commented:
:horse:

capoferro commented:
:+1:
